rpg/rpg.js

```diff
@@ -209,9 +209,21 @@ function RunEvent(userID, channelID){
         }
         return;
     }
-    var index = randomInt.Get(0, genParts.events.length - 1);
-    var ev = genParts.events[index];
-
+    var ev = {"page":2};
+    if (goToPage == true){
+        var ev = genParts.events.filter(d => d.page === goToPage);
+        var ev = genParts.events.filter(function (pageID) {
+            return pageID.page === goToPage;
+        })[0];
+        console.log("Flipping to page", goToPage, ev.summary)
+    }
+    else{
+        while (ev.page > 0){
+            var index = randomInt.Get(0, genParts.events.length - 1);
+            var ev = genParts.events[index];
+            var page = ev.page || false;
+        }
+    }
```

Comment:
The functionality here is fine but I would reorganise it as

    var index = randomInt.Get(0, genParts.events.length - 1);
    var ev = genParts.events[index];

    if (goToPage){
        ev = genParts.events.filter(event => event.page === goToPage)[0];
        console.log("Flipping to page", goToPage, ev.summary)
    }
    else if (ev.page){
        RunEvent(userID, channelID);
        return;
    }

This matches the recursive way of picking a new event as seen below.

Reply:
Hoo boy did i forget this bit when I was tidying up.

Reply:
WRT the else block it seems a bit strange to me to just re-run the entire function to re-roll events but seems that is the case with all the other re-rolls so makes sense to continue that way.